xcube/core/gen/iproc.py

```diff
@@ -101,6 +101,13 @@ def input_reader_params(self) -> dict:
         return dict()
 
     @abstractmethod
+    def get_time_for_sorting(self, dataset: xr.Dataset) -> Optional[str]:
+        """
+        Return a string of the datasets time for presorting the input list.
+        :param dataset: The dataset.
+        :return: The time string of the dataset or None.
+        """
+
```

Comment:
Why this new method? For sorting, `get_time_range` should be sufficient.
If there is a special need for this method, why it is abstract and why doesn't it have default implementation, e.g.

    def get_time_for_sorting(self, dataset: xr.Dataset) -> Optional[str]:
        t1, t2 = self.get_time_range()
        return (t1 + t2) / 2